File: android/app/src/main/java/jaggl/MapBuffer.java
package jaggl;

import jaclib.memory.NativeBuffer;

public final class MapBuffer extends NativeBuffer {

    public int c;

    public boolean a(int arg0, int arg1, int arg2) {
        if (this.c != 0) {
            return false;
        }
        long local9 = OpenGL.glMapBufferARB(arg0, arg2);
        if (local9 == 0L) {
            return false;
        } else {
            this.a(local9, arg1);
            this.c = arg0;
            return true;
        }
    }

    public boolean a() {
        return this.c != 0;
    }

    @Override
    public void a(byte[] arg0, int arg1, int arg2, int arg3) {
        if (this.c == 0) {
            throw new RuntimeException();
        }
        super.a(arg0, arg1, arg2, arg3);
    }

    public boolean b() {
        boolean local1 = true;
        if (this.c != 0) {
            local1 = OpenGL.glUnmapBufferARB(this.c);
            this.a(0L, 0);
            this.c = 0;
        }
        return local1;
    }
}

File: src/commonMain/kotlin/jaggl/MapBuffer.kt
package jaggl

import jaclib.memory.NativeBuffer

class MapBuffer : NativeBuffer() {
    private var c = 0

    public override fun a(arg0: ByteArray?, arg1: Int, arg2: Int, arg3: Int) {
        if (this.c == 0) {
            throw RuntimeException()
        }
        super.a(arg0, arg1, arg2, arg3)
    }

    fun a(arg0: Int, arg1: Int, arg2: Int): Boolean {
        if (this.c != 0) {
            return false
        }
        val local8: Long = OpenGL.Companion.glMapBufferARB(arg0, arg2)
        if (local8 == 0L) {
            return false
        } else {
            this.a(local8, arg1)
            this.c = arg0
            return true
        }
    }

    fun a(): Boolean {
        return this.c != 0
    }

    fun b(): Boolean {
        var local1 = true
        if (this.c != 0) {
            local1 = OpenGL.Companion.glUnmapBufferARB(this.c)
            this.a(0L, 0)
            this.c = 0
        }
        return local1
    }
}
